NASA-AMMOS/aerie-ui
# Custom tags input

Implements a custom tags input and integrates it into the new tag architecture for testing purposes. Partially addresses #605 
<img width="551" alt="image" src="https://github.com/NASA-AMMOS/aerie-ui/assets/4419607/af22cfe8-6b56-4ce6-b05c-4d92acdb4818">

Testing:
1. Open a plan
2. Open up activity directive form
3. Add existing tags from the `tags` field by typing into the tags input and selecting existing tags using arrow keys or via click. 
4. Add new tags by entering a tag that does not yet exist and hitting enter or clicking on "add <tag_name> (enter)"
5. Disassociate tags from the activity directive by deleting them from the tags input. This can be accomplished by focusing on the input and pressing backspace/delete (windows/mac) or by clicking on the tags in the input.
6. Click on a different activity directive and ensure that the tags you have added from the other directive are available for selection.

Questions:
- Pastel colors are automatically generated for new tags. Is this behavior acceptable if we don't ship tag color editing at the same time? Alternatively we use a single color for now and/or users can edit tag colors via API.
- Should the Tags component be more generic? It could potentially serve as a general autosuggest with single/multiple inputs and could be decoupled from Aerie's tag type? Will require more work and investigation if desired.

**Not ready to merge, waiting on https://github.com/NASA-AMMOS/aerie/pull/1008**
